src/consensus/proposer.rs

```diff
@@ -489,6 +490,14 @@ impl Proposer for BlockProposer {
             .collect_confirmed_shard_witnesses(height, version, timeout)
             .await;
 
+        let mempool_timeout = Duration::from_millis(200);
+        let messages = self
```

Comment:
Nothing in the mempool currently gets routed to shard 0. Maybe worth just defaulting to an empty list of transactions rather than having this in here. 

We should test that the engine code works with non-empty transactions though.